libraries/AC_Avoidance/AP_OABendyRuler.cpp

```diff
@@ -112,13 +155,18 @@ bool AP_OABendyRuler::update(const Location& current_loc, const Location& destin
                     // calculate minimum margin to fence and obstacles for this scenario
                     float margin2 = calc_avoidance_margin(test_loc, test_loc2);
                     if (margin2 > _margin_max) {
+                        // if the chosen direction is directly towards the destination avoidace can be turned off 
+                        const bool active = (i != 0 || j != 0);
+                        float final_bearing = bearing_test;
+                        float final_margin = margin;
+                        //check if we need avoid test_bearing and continue on previous bearing
+                        const bool avoid_change = avoid_bearing_change(destination, current_loc, active, bearing_test, lookahead_step1_dist, margin, final_bearing, final_margin);
```

Comment:
let's not use the word, "avoid_" for this variable and related function because it's too easy to mix up with "avoidance" (i.e. avoiding an object).  Instead let's call it "ignore_change" or "ignore_bearing_change".